main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -296,13 +311,33 @@ object GenExpression {
       visitor.visitVarInsn(iStore, sym.getStackOffset + 1)
 
       // Compile the scope body
+      visitor.visitLabel(beforeTryBlock)
       compileExpression(exp, visitor, currentClass, lenv0, entryPoint)
 
       // When we exit the scope, call the region's `exit` method
       val iLoad = AsmOps.getLoadInstruction(JvmType.Reference(BackendObjType.Region.jvmName))
       visitor.visitVarInsn(iLoad, sym.getStackOffset + 1)
       visitor.visitMethodInsn(INVOKEVIRTUAL, BackendObjType.Region.jvmName.toInternalName, BackendObjType.Region.ExitMethod.name, 
         BackendObjType.Region.ExitMethod.d.toDescriptor, false)
+      visitor.visitLabel(afterTryBlock)
+
+      // Compile the finally block which gets called if no exception is thrown
+      visitor.visitVarInsn(iLoad, sym.getStackOffset + 1)
+      visitor.visitMethodInsn(INVOKEVIRTUAL, BackendObjType.Region.jvmName.toInternalName, BackendObjType.Region.ReThrowChildExceptionMethod.name,
+        BackendObjType.Region.ReThrowChildExceptionMethod.d.toDescriptor, false)
+      visitor.visitJumpInsn(GOTO, afterFinally)
+
+      // Compile the finally block which gets called if an exception is thrown
+      visitor.visitLabel(finallyBlock)
+      visitor.visitVarInsn(AsmOps.getStoreInstruction(JvmType.Reference(JvmName.Throwable)),
+        sym.getStackOffset + 2)
```

Comment:
What is the reason for `+ 2` here? Isn't that assuming that the stack beyond `sym` is unused?

Reply:
If I understand correctly, local variables are stored at offsets from `sym.getStackOffset`. So this code has two local variables: one which stores the region (at `sym.getStackOffset + 1`) and one which stores the exception (at `sym.getStackOffset + 2`).

But actually, I think I can do away with it altogether: I was copying the code generated by `javac`, but I don't actually need to store the variable at all, which makes things simpler.